applications/FluidDynamicsApplication/custom_constitutive/newtonian_two_fluid_3d_law.cpp

```diff
@@ -78,6 +79,13 @@ double NewtonianTwoFluid3DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters
     double viscosity;
     EvaluateInPoint(viscosity, DYNAMIC_VISCOSITY, rParameters);
     const Properties& prop = rParameters.GetMaterialProperties();
+    const GeometryType &rGeom = rParameters.GetElementGeometry();
+
+    if (rGeom.Has(ARTIFICIAL_DYNAMIC_VISCOSITY))
+    {
+        double artificial_visocisty = rGeom.GetValue(ARTIFICIAL_DYNAMIC_VISCOSITY);
+        viscosity += artificial_visocisty;
```

Comment:
```suggestion
        viscosity += rGeom.GetValue(ARTIFICIAL_DYNAMIC_VISCOSITY);
```
There is no need to allocate the temporary value (indeed, the compiler optimizer is probably doing this for you).